server/controllers/operations.coffee

```diff
@@ -42,50 +42,73 @@ module.exports.delete = (req, res) ->
+                        callback null
+
+                    # text search
+                    else if paramSearchText? and paramSearchText isnt "" and \
+                            title.search(paramQueryText) < 0
+                        callback null
+
+                    # the right one
+                    else
+                        callback null, operation
+
+                # check all bank operations
+                async.concat operations, treatment, (err, results) ->
+                    if err?
+                        errorMsg = 'Server error occurred while retrieving data'
+                        res.send 500, error: errorMsg
+                    else
+                        res.send 200, results
+
+
+module.exports.file = (req, res, next) ->
+    binaryPath = "/data/#{req.params.bankOperationID}/binaries/file"
+
+    id = process.env.NAME
+    pwd = process.env.TOKEN
+    basic = "Basic #{new Buffer("#{id}:#{pwd}").toString('base64')}"
+    options =
+        host: 'localhost'
+        port: 9101
+        path: binaryPath
+        headers:
+            Authorization: basic
+
+    request = require('http').get options, (stream) ->
+
+        if stream.statusCode is 200
+            res.on 'close', -> request.abort()
+            stream.pipe res
         else
-            paramDateFrom =  new Date req.body.dateFrom
-            paramDateTo = new Date req.body.dateTo
-            paramAmountFrom =  Number req.body.amountFrom
-            paramAmountTo = Number req.body.amountTo
-            paramSearchText = req.body.searchText
-            async = require "async"
-
-            treatment = (operation, callback) ->
-                # apply filters to dermine if the operation should be returned
-                amount = Number operation.amount
-                date = new Date operation.date
-                title = operation.title.toLocaleUpperCase()
-                paramQueryText = paramSearchText.toLocaleUpperCase()
-
-                # dates
-                if date < paramDateFrom or date > paramDateTo
-                    callback null
-
-                # amounts
-                else if amount < paramAmountFrom or amount > paramAmountTo
-                    callback null
-
-                # text search
-                else if paramSearchText? and paramSearchText isnt "" and \
-                        title.search(paramQueryText) < 0
-                    callback null
-
-                # the right one
-                else
-                    callback null, operation
-
-            # check all bank operations
-            async.concat operations, treatment, (err, results) ->
-                if err?
-                    errorMsg = 'Server error occurred while retrieving data'
-                    res.send 500, error: errorMsg
-                else
-                    res.send 200, results
+            res.status(404).send 'File not found'
```

Comment:
Is the only possible error that the file isn't found? Could we just `res.status(stream.statusCode);` instead?